src/ophyd_async/epics/_backend/_aioca.py

```diff
@@ -88,11 +88,15 @@ def write_value(self, value: Union[Enum, str]):
             return value
 
     def value(self, value: AugmentedValue):
-        return self.enum_class(value)
+        return value
```

Comment:
This means that:
```python
class MyEnum(str, Enum):
   aaa = "AAA"
   bbb = "BBB"

sig = epics_signal_rw(MyEnum, pv)
await sig.connect()
assert (await sig.get_value()) == MyEnum.aaa
assert (await sig.get_value()) == "AAA"
```
Unfortunately
```python
print(await sig.get_value())  # prints "AAA" not <MyEnum.aaa: "AAA">
```
@DominicOram is this ok?

Reply:
Added some tests to ensure that:
for signals backed by an enum:

Signal[str].get_value() is not Enum.val
Signal[Enum].get_value() is Enum.val

in each case, Signal.get_value() == "val" and Signal.get_value() == Enum.val